src/lib/tpm.c

```diff
@@ -22,13 +24,24 @@
 #include <tss2/tss2_esys.h>
 #include <tss2/tss2_mu.h>
 
+#include "checks.h"
 #include "openssl_compat.h"
 #include "pkcs11.h"
 #include "log.h"
 #include "mutex.h"
 #include "tcti_ldr.h"
 #include "tpm.h"
 
+/**
+ * Maps 4-byte manufacturer identifier to manufacturer name.
+ */
+static const char *TPM2_MANUFACTURER_MAP[][2] = {
+    {"INTC", "Intel"},
+    {"IFX ", "Infineon"},
+    {"IBM ", "IBM"},
```

Comment:
expand the map to the best of your ability to the registry and ensure a default handler to just plop their data in there.

Reply:
The default *is* handled in the current code.
By expanding you mean adding the other entries from tcg?

Reply:
I updated it to the tpm2.0 vendors known to me.